shell/core/types-provisioning.ts

```diff
@@ -102,7 +102,7 @@ export interface IClusterProvisioner {
   group?: string;
 
   /**
-   * Disable the cluster provider card
+   * Hide the cluster provider card
```

Comment:
I don't think we should switch the functionality of an existing function.

We have a couple of different options
- Get providers to declare they are a kontainer driver and the kontainer driver name. We would then do the check dashboard side. JSDOC / dev kit docs / release notes should make it clear that we won't show the provider unless we find a matching kontainer with the driver name.
  - We could chose to make this optional, so only when supplied and we don't find a matching kontainer driver do we hide the option 
- Add a specific `hide` provider prop

Reply:
>Add a specific hide provider prop

I think this is more likely to be useful to other extension developers than adding something so kontainer-specific, so I'll try this approach